sdk/src/main/java/com/mapbox/maps/MapView.kt

```diff
@@ -69,39 +69,25 @@ open class MapView : FrameLayout, MapPluginProviderDelegate, MapControllable {
   )
 
   /**
-   * Build a [MapView] with a [Context] object, a [AttributeSet] object, an
-   * [Int] which represents a style attribute file, and an [Int] which represents a
-   * style resource file.
-   */
-  @TargetApi(Build.VERSION_CODES.LOLLIPOP)
-  constructor(
-    context: Context,
-    attrs: AttributeSet?,
-    defStyleAttr: Int,
-    defStyleRes: Int
-  ) : this(
-    context,
-    MapboxMapOptions(context, context.resources.displayMetrics.density, attrs),
-    attrs,
-    defStyleAttr,
-    defStyleRes
-  )
-
-  /**
-   * Build a [MapView] with a [Context] object, a [MapboxMapOptions] object,
+   * Build a [MapView] with a [Context] object, a [MapInitOptions] object,
    * a [AttributeSet] object, an [Int] which represents a style attribute file,
    * and an [Int] which represents a style resource file.
    */
   @Suppress("UNCHECKED_CAST")
   @TargetApi(Build.VERSION_CODES.LOLLIPOP)
   constructor(
     context: Context,
-    options: MapboxMapOptions,
     attrs: AttributeSet?,
     defStyleAttr: Int,
-    defStyleRes: Int
+    defStyleRes: Int,
+    initOptions: MapInitOptions = MapInitOptions(context),
   ) : super(context, attrs, defStyleAttr, defStyleRes) {
-    val view = if (options.textureView) {
+    val resolvedMapInitOptions = if (attrs != null) {
+      parseTypedArray(context, attrs)
```

Comment:
Does this mean that the options defined in xml attributes will overwrite the ones defined by user programmatically?

Reply:
Good catch, need to rethink about it.